File: velolabs_repos/operator-android/domain/src/main/java/io/lattis/domain/models/Fleet.kt
package io.lattis.domain.models

import com.google.gson.annotations.SerializedName

data class Fleet (
    @SerializedName("vehiclesCount")
    var vehiclesCount:Int?=null,

    @SerializedName("legal")
    var legal:String?=null,

    @SerializedName("type")
    var type:String?=null,

    @SerializedName("name")
    var name:String?=null,

    @SerializedName("logo")
    var logo:String?=null,

    @SerializedName("id")
    var id:Int?=null,

    @SerializedName("address")
    var address:Address?=null
){

    data class Address(
        @SerializedName("city")
        var city:String?=null,

        @SerializedName("country")
        var country:String?=null,

        @SerializedName("id")
        var id:Int?=null,

        @SerializedName("postalCode")
        var postalCode:String?=null,

        @SerializedName("address1")
        var address1:String?=null,

        @SerializedName("state")
        var state:String?=null,

        @SerializedName("address2")
        var address2:String?=null
    )

}

File: velolabs_repos/operator-ios/Shared/Model/Fleet.swift
//
//  Fleet.swift
//  Operator
//
//  Created by Ravil Khusainov on 23.02.2021.
//

import SwiftUI

struct Fleet: Codable, Identifiable, Hashable {
    let id: Int
    let name: String?
    let email: String?
    let logo: URL?
    let address: Address?
    let vehiclesCount: Int
}

extension Fleet {
    var fullAddress: String? {
        guard let address = address else { return nil }
        var addr = address.address2 ?? ""
        let devider = ", "
        if let st = address.address1 {
            addr += devider + st
        }
        if let city = address.city {
            addr += devider + city
        }
        if let state = address.state {
            addr += devider + state
        }
        if let country = address.country {
            addr += devider + country
        }
        if let code = address.postalCode {
            addr += devider + code
        }
        return addr.trimmingCharacters(in: .init(charactersIn: devider))
    }
}

